arch/arm64/src/Toolchain.defs

```diff
@@ -247,5 +247,11 @@ LDMODULEFLAGS = -r -T $(call CONVERT_PATH,$(TOPDIR)/libs/libc/modlib/gnu-elf.ld)
 CELFFLAGS = $(CFLAGS) -fvisibility=hidden -mlong-calls # --target1-abs
 CXXELFFLAGS = $(CXXFLAGS) -fvisibility=hidden -mlong-calls # --target1-abs
 
-LDELFFLAGS = -r -e main
+LDELFFLAGS = -r -e __start
```

Comment:
In flat build will this not result in multiple definitions of __start (due to the OS entrypoint) ?

Reply:
I think main is more reasonable, is there a more elegant way?

Reply:
the real entry point isn't main in popular libc implementation: https://en.wikipedia.org/wiki/Crt0

Reply:
just apps build with elf, app entry will be __start, __start will init c++ constructor and call main

Reply:
I think _start / __start is fine as user app entrypoint, this is how crt typically works.